IdentityCore/src/cache/MSIDTokenCache.h

```diff
@@ -25,33 +25,86 @@
 
 @class MSIDToken;
 @class MSIDTokenCacheKey;
+@class MSIDUser;
+
+@protocol MSIDTokenCacheDataSource;
 
 @interface MSIDTokenCache : NSObject
 
-- (BOOL)saveToken:(MSIDToken *)token
-           forKey:(MSIDTokenCacheKey *)key
-            error:(NSError **)error;
+- (id)initWithDataSource:(id<MSIDTokenCacheDataSource>)dataSource;
+- (id<MSIDTokenCacheDataSource>)dataSource;
 
-- (MSIDToken *)getAdalAT:(NSString *)userId
-                resource:(NSString *)resource
-                clientId:(NSString *)clientId
-                   error:(NSError **)error;
+/*!
+ Returns a AT/RT Token Cache Item for the given parameters. The RT in this item will only be good
+ for the given resource. If no RT is returned in the item then a MRRT or FRT should be used (if
+ available).
+ */
+- (MSIDToken *)getAdalATRTforUser:(MSIDUser *)user
+                        authority:(NSURL *)authority
+                         resource:(NSString *)resource
+                         clientId:(NSString *)clientId
+                          context:(id<MSIDRequestContext>)context
+                            error:(NSError **)error;
 
-- (MSIDToken *)getMsalAT:(NSString *)userId
-                  scopes:(NSSet<NSString *> *)scopes
-                clientId:(NSString *)clientId
-                   error:(NSError **)error;
+- (MSIDToken *)getFRTforUser:(MSIDUser *)user
+                   authority:(NSURL *)authority
+                    familyId:(NSString *)familyId
+                     context:(id<MSIDRequestContext>)context
+                       error:(NSError **)error;
 
-- (MSIDToken *)getFRT:(NSString *)userId
-             familyId:(NSString *)familyId
-                error:(NSError **)error;
+- (MSIDToken *)getAdfsUserTokenForAuthority:(NSURL *)authority
```

Comment:
if we keep this one, it should return a different type of token that contains both AT and RT